apispec/ext/marshmallow/swagger.py

```diff
@@ -48,211 +50,15 @@
 CUSTOM_FIELD_MAPPING_ATTR = '__swagger_field_mapping'
 
 
-def map_to_swagger_type(*args):
```

Comment:
This was part of the public API. Perhaps this should be exposed `MarshmallowPlugin` as an instance method?

Reply:
It is already exposed. Sort of. It is in the `Swagger` object.

```python
ma_plugin  = MarshmallowPlugin()

APISpec(
        title='My super API',
        version='0.1',
        plugins=(ma_plugin, ),
        openapi_version='3.0.1')

@ma_plugin.swagger.map_to_swagger_type('string', 'uuid')
class UuidField(StrField):
    [...]
```

I could add a `map_to_swagger_type` method to `MarshmallowPlugin` to proxy this one, but unless I add complicated or awkward logic, it would still have to be called after registration in `APISpec` instance, as this is when the `Swagger` instance is created.

I can't think of any simpler way to do this. We could create the `Swagger` object at init of the MA plugin, but it still wouldn't be totally usable until a spec object is set (just like the MA plugin) anyway.

Or maybe we can merge the MA plugin and the swagger object. Using composition, we'd keep them in separate files.

```python
class MarshmallowPlugin(BasePlugin, Swagger):
    """APISpec plugin handling marshmallow schemas
    [...]
    """
```

This way, all methods are accessible at MA plugin level.

Looks weird to me, but perhaps because I too rarely use composition.